lib/srv/usermgmt.go

```diff
@@ -394,29 +426,21 @@ func (u *HostUserManagement) UpsertUser(name string, ui services.HostUsersInfo)
 	}
 	ui.Groups = groups
 
-	if err := u.ensureGroupsExist(types.TeleportDropGroup, types.TeleportKeepGroup); err != nil {
+	if err := u.ensureGroupsExist(types.TeleportDropGroup, types.TeleportKeepGroup, types.TeleportStaticGroup); err != nil {
 		return nil, trace.WrapWithMessage(err, "creating teleport system groups")
 	}
 
 	if err := u.ensureGroupsExist(groups...); err != nil {
 		return nil, trace.WrapWithMessage(err, "creating configured groups")
 	}
 
-	var closer io.Closer
-	if ui.Mode == types.CreateHostUserMode_HOST_USER_MODE_INSECURE_DROP {
-		closer = &userCloser{
-			username: name,
-			users:    u,
-			backend:  u.backend,
-		}
-	}
-
-	if err := u.updateUser(name, ui); err != nil {
+	closer, err := u.updateUser(name, ui)
```

Comment:
I may be missing a difference, but it looks like the behavior for closers is still the same with successfully updated/created users returning a closer when in `insecure-drop` mode. Any reason for pushing closer generation into the `create` and `update` functions instead of leaving it in `UpsertUser`?

Reply:
When `keep` and `insecure-drop` were the only modes and we can freely swap between them, knowing the mode you want is enough to know ahead of time if you need a closer. Now, you might for example try to update a static user to `insecure-drop`, and only then realize that you shouldn't have a closer.